app/src/main/java/org/novinger/jason/demoreader/ArticleAdapter.java

```diff
@@ -0,0 +1,55 @@
+package org.novinger.jason.demoreader;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.novinger.jason.demoreader.datamodels.Article;
+
+import java.util.ArrayList;
+
+/**
+ * Created by jason on 5/1/16.
+ */
+public class ArticleAdapter extends RecyclerView.Adapter<ArticleAdapter.ViewHolder> {
+    private ArrayList<Article> mArticles;
+
+    public static class ViewHolder extends RecyclerView.ViewHolder {
```

Comment:
ok, let me know if you want the long story...  it came about because of bad habits and slow xml parsing, but is now baked in the cake, so i think turned out maybe a little differently than it might have without those bumps